utils/near-cache/src/lib.rs

```diff
@@ -19,17 +19,29 @@ where
     }
 
     /// Gets value for `key`
-    pub fn get_or_insert(&self, key: K, f: impl FnOnce(&K) -> V) -> V
+    pub fn get_or_insert<F>(&self, key: K, f: F) -> V
     where
         V: Clone,
+        F: FnOnce(&K) -> V,
+    {
+        match self.get_or_try_insert(key, |k| Ok::<V, std::convert::Infallible>(f(k))) {
+            Ok(it) => it,
+            Err(never) => match never {},
```

Comment:
hmm isn't this equivalent to `Err(_) => ()`?

Reply:
not really, the type of `()` is `()`, the type of `match never {}` is `!`. The result type here is `V`, and `match never {}` is compatible with that. 

For context: `Infallible` is an empty enum: `enum Infallible {}`

Reply:
Why not use:
```
diff --git i/utils/near-cache/src/lib.rs w/utils/near-cache/src/lib.rs
index 069662c90..c13416021 100644
--- i/utils/near-cache/src/lib.rs
+++ w/utils/near-cache/src/lib.rs
@@ -26,10 +26,7 @@ where
         V: Clone,
         F: FnOnce(&K) -> V,
     {
-        match self.get_or_try_put(key, |k| Ok::<V, std::convert::Infallible>(f(k))) {
-            Ok(it) => it,
-            Err(never) => match never {},
-        }
+        self.get_or_try_put(key, |k| Ok::<V, std::convert::Infallible>(f(k))).unwrap()
     }
 
     /// Returns the value of they key in the cache if present, otherwise
```

Reply:
`.unwrap()` is an operation that can in general fail at runtime. In this case it can't actually fail, but you need to read the context to understand this. In contrast, `match never {}` can't fail in a statically-checked way. 

Really, thish should be just `Result::into_ok` (https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.into_ok) but that needs to wait until `!` is stabilized properly.

Reply:
In previous codebases I used

```
Result::<_, Infallible>::unwrap(result)
```

to very signal clearly enough that “yeah this is unwrap, but no it is actually into_ok”. At codegen level LLVM can see easily enough that panic within `unwrap` is not reachable panic won't actually materialize.

Reply:
Excellent suggestion, thanks!